deep_qa/data/vocabulary.py

```diff
@@ -7,35 +7,38 @@
 
 logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
 
+class _NamespaceDependentDefaultDict(defaultdict):
```

Comment:
A short comment here explaining this would have helped me: "A defaultdict which initialises different default dictionaries depending on whether or not the name_space requires default padding and unknown tokens."